### DISCUSSION TITLE
Interactive graph view as Obsidian for a Quarto website

### DISCUSSION THREAD

**Discussion by pipme** [quarto-dev/quarto-cli#5901] [Feature Requests]
### Description

It would be nice to have an option (or maybe an extension) to add an interactive graph for visualizing linked `qmd` files (HTML pages).
 
https://help.obsidian.md/Plugins/Graph+view
![image](https://github.com/quarto-dev/quarto-cli/assets/30783979/6782dde0-8366-40aa-9588-b6750633e33b)

**Comment by ning3739** [comment]
Hello Pipme, have you heard of [**echarts4r**](https://echarts4r.john-coene.com/articles/chart_types.html#tree)? It might be just what you're searching for! Alternatively, if you're familiar with JavaScript, you could explore [**echarts**](https://echarts.apache.org/examples/en/editor.html?c=graph-force).

**Reply by ning3739** [reply]
https://echarts4r.john-coene.com/articles/graph.html

**Reply by mcanouil** [reply]
@NING3739 Thanks for the suggestion, although it does not compute nodes and edges for pages interdependencies.

I don't think the structure scheme is stored somewhere (or can be) when Quarto resolves this.
@cscheid might know better if it is doable to have the pages dependency graph exposed to user.

**Reply by cscheid** [reply]
@mcanouil not before 1.5 is out. 1.5 is not the upcoming release, but the _next_ one, which will include a lot of dependency management work.

**Reply by mcanouil** [reply]
Thanks, it's what I had in mind from an old answer or yours in a GitHub Discussion.

**Reply by pipme** [reply]
Thanks for the replies. For reference, related issue #1585.

**Reply by haxfn** [reply]
this feature seems cool!

**Comment by emstruong** [comment]
I agree that this feature would be cool. Quartz apparently does a similar thing and I think this would be a great addition to how Quarto renders HTMLs: https://quartz.jzhao.xyz/features/graph-view

I suppose it is related to #1585

**Reply by haxfn** [reply]
Wow, quartz, I never knew about this..
